examples/colors/store.rs

```diff
@@ -0,0 +1,193 @@
+                        ParamFlags::READWRITE,
+                    ),
+                ]
+            });
+            PROPERTIES.as_ref()
+        }
+
+        fn set_property(&self, _obj: &Self::Type, _id: usize, value: &Value, pspec: &ParamSpec) {
+            match pspec.name() {
+                "size" => self.set_size(_obj, value.get().unwrap()),
+                "limit" => {
+                    self.limit.replace(value.get().unwrap());
+                }
+                _ => unimplemented!(),
+            }
+        }
+
+        fn property(&self, _obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> Value {
+            match pspec.name() {
+                "size" => (self.size.borrow()).to_value(),
+                "limit" => (self.limit.borrow()).to_value(),
+                _ => unimplemented!(),
+            }
+        }
+    }
+
+    impl ColorStore {
+        pub fn set_size(&self, _obj: &super::ColorStore, size: u32) {
+            let old_size = *self.size.borrow();
+
+            *self.size.borrow_mut() = size;
+
+            if size > old_size {
+                _obj.items_changed(old_size, 0, size - old_size);
+            } else if old_size > size {
+                _obj.items_changed(size, old_size - size, 0);
+            }
+
+            _obj.notify_by_pspec(&_obj.list_properties()[0]);
+        }
+    }
+
+    impl ListModelImpl for ColorStore {
+        fn item_type(&self, _: &Self::Type) -> glib::Type {
+            Color::static_type()
+        }
+
+        fn n_items(&self, _: &Self::Type) -> u32 {
+            *self.size.borrow() as u32
+        }
+
+        fn item(&self, _: &Self::Type, position: u32) -> Option<glib::Object> {
+            let map = [
+                0xFF0000, 0x00FF00, 0x0000FF, 0x7F0000, 0x007F00, 0x00007F, 0x3F0000, 0x003F00,
+                0x00003F, 0x1F0000, 0x001F00, 0x00001F, 0x0F0000, 0x000F00, 0x00000F, 0x070000,
+                0x000700, 0x000007, 0x030000, 0x000300, 0x000003, 0x010000, 0x000100, 0x000001,
+            ];
+
+            let mut result = 0;
+
+            for i in 0..map.len() {
+                if position & (1 << i) != 0 {
+                    result ^= map[i];
+                }
+            }
+
+            let color = Color::new(gtk::gdk::RGBA::new(
+                ((result >> 16) & 0xFF) as f32 / 255.,
+                ((result >> 8) & 0xFF) as f32 / 255.,
+                (result & 0xFF) as f32 / 255.,
+                1.0,
+            ));
+
+            Some(color.clone().upcast::<glib::Object>())
+        }
+    }
+}
+
+glib::wrapper! {
+    pub struct ColorStore(ObjectSubclass<imp::ColorStore>) @implements gtk::gio::ListModel;
+}
+
+impl ColorStore {
+    pub fn new(size: u32) -> Self {
+        glib::Object::new(&[("size", &size.to_value())]).expect("Failed to create `ColorStore`.")
+    }
+
+    pub fn size(&self) -> u32 {
+        self.property::<u32>("size")
+    }
+
+    pub fn limit(&self) -> u32 {
+        self.property::<u32>("limit")
+    }
```

Comment:
```suggestion
    pub fn limit(&self) -> u32 {
        self.property("limit")
    }
```
pretty sure you can just write :-)